chain/network/src/peer_manager/tests/sync_snapshot_hosts.rs

```diff
@@ -0,0 +1,282 @@
+use crate::network_protocol::testonly as data;
+use crate::network_protocol::SnapshotHostInfo;
+use crate::network_protocol::SyncSnapshotHosts;
+use crate::peer;
+use crate::peer_manager;
+use crate::peer_manager::peer_manager_actor::Event as PME;
+use crate::tcp;
+use crate::testonly::{make_rng, AsSet as _};
+use crate::types::NetworkRequests;
+use crate::types::PeerManagerMessageRequest;
+use crate::types::PeerMessage;
+use near_async::time;
+use near_crypto::SecretKey;
+use near_o11y::testonly::init_test_logger;
+use near_o11y::WithSpanContextExt;
+use near_primitives::hash::CryptoHash;
+use near_primitives::network::PeerId;
+use near_primitives::types::EpochHeight;
+use near_primitives::types::ShardId;
+use pretty_assertions::assert_eq;
+use std::sync::Arc;
+
+/// Create an instance of SnapshotHostInfo for testing purposes
+fn make_snapshot_host_info(
+    peer_id: &PeerId,
+    epoch_height: EpochHeight,
+    shards: Vec<ShardId>,
+    secret_key: &SecretKey,
+) -> Arc<SnapshotHostInfo> {
+    let sync_hash = CryptoHash::hash_borsh(epoch_height);
+    Arc::new(SnapshotHostInfo::new(peer_id.clone(), sync_hash, epoch_height, shards, secret_key))
+}
+
+/// Used to consume peer events until there's an event of type SyncSnapshotHosts
+fn take_sync_snapshot_msg(event: crate::peer::testonly::Event) -> Option<SyncSnapshotHosts> {
+    match event {
+        peer::testonly::Event::Network(PME::MessageProcessed(
+            tcp::Tier::T2,
+            PeerMessage::SyncSnapshotHosts(msg),
+        )) => Some(msg),
+        _ => None,
+    }
+}
+
+/// Used to consume peer manager events until there's an event of type SyncSnapshotHosts
+fn take_sync_snapshot_msg_manager(
```

Comment:
See comment at the end of the file, I think this lies in is a weird middle ground between consuming just the network events (as done in the first tests with fake peer managers) and what is possible in the last test (in the full peer managers there exists a SnapshotHostsCache instance and we can verify its state).